### PR TITLE
Add `rtables.officer` as a dependency

### PR DESCRIPTION
**Pull request by vedhav**
Changes:
1. Updates the `rtables` version
2. Adds `rtables.officer` as a new dependency

Note that this blocks CRAN release of `teal.reporter` untill the `rtables.officer` and `rtables` is released.

### CODE REVIEW

File: staged_dependencies.yaml [hunk 1]
```diff
@@ -25,6 +25,9 @@ upstream_repos:
   insightsengineering/rtables:
     repo: insightsengineering/rtables
     host: https://github.com
+  insightsengineering/rtables.officer:
+    repo: insightsengineering/rtables.officer
+    host: https://github.com
```
Comment (shajoezhu) [inline staged_dependencies.yaml:30]: i thought you guys have stopped using staged.dependnecies?

**Review by shajoezhu — APPROVED** [review]
thanks a lot @vedhav for the changes, I think it all looks good. i was wondering if the new changes in rtables, is blocking teal package development? If not, can we work together with the release plan. Thanks!

File: staged_dependencies.yaml (hunk 1 above)
Comment (vedhav) [reply staged_dependencies.yaml:30]: That's correct, we're in the middle of migrating ([so far only teal is migrated](https://github.com/insightsengineering/nestdevs-tasks/issues/65)), But, we're working on other tasks so it is on hold until we get to it.

**Comment by vedhav** [comment]
Thanks @shajoezhu we do not plan to release the `teal.reporter` this increment so we can follow the `rtables` release plan.

**Pull request merged**